ldb/operational_test.go

```diff
@@ -39,24 +33,167 @@ func testOperationalMode(t *testing.T) {
 	_ = os.RemoveAll(dbnamever)
 	db, err := btcdb.CreateDB("leveldb", dbname)
 	if err != nil {
-		t.Errorf("Failed to open test database %v", err)
-		return
+		return nil, nil, "", "", err
 	}
-	defer os.RemoveAll(dbname)
-	defer os.RemoveAll(dbnamever)
-	defer func() {
-		if err := db.Close(); err != nil {
-			t.Errorf("Close: unexpected error: %v", err)
-		}
-	}()
 
 	testdatafile := filepath.Join("..", "testdata", "blocks1-256.bz2")
 	blocks, err := loadBlocks(t, testdatafile)
 	if err != nil {
-		t.Errorf("Unable to load blocks from test data: %v", err)
+		return nil, nil, "", "", err
+	}
+
+	return db, blocks, dbname, dbnamever, nil
+}
+
+func TestOperational(t *testing.T) {
+	testOperationalMode(t)
+}
+
+// testAddrIndexOperations ensures that all normal operations concerning
+// the optional address index function correctly.
+func testAddrIndexOperations(t *testing.T, db btcdb.Db, newestBlock *btcutil.Block, newestSha *btcwire.ShaHash, newestBlockIdx int64) {
+	// Metadata about the current addr index state should be unset.
+	sha, height, err := db.FetchAddrIndexTip()
+	if err != btcdb.ErrAddrIndexDoesNotExist {
+		t.Errorf("Address index metadata shouldn't be in db, hasn't" +
+			" be built up yet.")
+		return
```

Comment:
It may not be wanted here as this is a helper function for the exported Test functions, but you can use t.Fatal or t.Fatalf instead of Error/Errorf and returning to immediately terminate a test after an error.